### PR TITLE
Add Badging API GroupData & Init documentations for `WorkerNavigator.clearAppBadge` and `WorkerNavigator.setAppBadge`

### LINKED ISSUE #22441: Some APIs are not listed while saying "a list of all the APIs"
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API

### What specific section or headline is this issue about?

Specifications

### What information was incorrect, unhelpful, or incomplete?

The page stats **all the APIs** in

> This is a list of all the APIs that are available.

However, there are some APIs that aren't listed up. Some examples of such unlisted APIs are:

* [Web NFC API - Web APIs | MDN](https://developer.mozilla.org/en-US/docs/Web/API/Web_NFC_API)
* [Web Serial API - Web APIs | MDN](https://developer.mozilla.org/en-US/docs/Web/API/Web_Serial_API)

### What did you expect to see?

Actually all APIs are listed up.

At least all APIs that have pages in MDN are listed up.

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/index.md
* Last commit: https://github.com/mdn/content/commit/6fefcdd237a377af5c066dc2734c118feadbbef9
* Document last modified: 2022-09-02T12:08:42.000Z

</details>

### LINKED ISSUE #29853: Bug of `{{DefaultAPISidebar("Badging API")}}`
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/API/Badging_API

### What specific section or headline is this issue about?

_No response_

### What information was incorrect, unhelpful, or incomplete?

the {{DefaultAPISidebar("Badging API")}} is completely empty without anything, whether the api summary or its methods

### What did you expect to see?

ditto

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/api/badging_api`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/API/Badging_API
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/api/badging_api/index.md
* Last commit: https://github.com/mdn/content/commit/835d6632d59993861a0458510402787f8a2c3cb3
* Document last modified: 2023-10-25T17:27:32.000Z

</details>

### PR DESCRIPTION
**Pull request by skyclouds2001**
### Description

add Badging API to group data

init `WorkerNavigator.clearAppBadge()` and `WorkerNavigator.setAppBadge()` documentation

also remove `{{AvailableInWorkers}}` from `Navigator.clearAppBadge()` and `Navigator.setAppBadge()`

also remove extra context of `navigator.setClientBadge()` and `navigator.clearClientBadge()`, which not removed completely in #29851 

### Related issues and pull requests

Fixes #29853

this is a part of #22441 , and could Fixes #22441 since current api is the only api that haven't fixed in #22441

### CODE REVIEW

File: files/jsondata/GroupData.json [hunk 1]
```diff
@@ -39,6 +39,13 @@
       "properties": [],
       "events": []
     },
+    "Badging API": {
+      "overview": ["Badging API"],
+      "interfaces": [],
+      "methods": ["Navigator.clearAppBadge()", "Navigator.setAppBadge()"],
```
Comment (wbamberg) [inline files/jsondata/GroupData.json:45]: The spec adds these to `WorkerNavigator` as well (https://w3c.github.io/badging/#extensions-to-the-navigator-and-workernavigator-interfaces) - we don't have pages for them there, but we should have.

**Comment by skyclouds2001** [comment]
And compat data for `WorkerNavigator.clearAppBadge()` and `WorkerNavigator.setAppBadge()` need to fix by https://github.com/mdn/browser-compat-data/issues/21177

File: files/en-us/web/api/workernavigator/clearappbadge/index.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [Badging API](/en-US/docs/Web/API/Badging_API)
+- [Badging for app icons](https://web.dev/badging-api/)
```
Comment (skyclouds2001) [inline files/en-us/web/api/workernavigator/clearappbadge/index.md:55]:
```suggestion
- [Badging for app icons](https://developer.chrome.com/docs/capabilities/web-apis/badging-api/)
```

File: files/en-us/web/api/workernavigator/setappbadge/index.md [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [Badging API](/en-US/docs/Web/API/Badging_API)
+- [Badging for app icons](https://web.dev/badging-api/)
```
Comment (skyclouds2001) [inline files/en-us/web/api/workernavigator/setappbadge/index.md:58]:
```suggestion
- [Badging for app icons](https://developer.chrome.com/docs/capabilities/web-apis/badging-api/)
```

File: files/en-us/web/api/workernavigator/clearappbadge/index.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [Badging API](/en-US/docs/Web/API/Badging_API)
```
Comment (skyclouds2001) [inline files/en-us/web/api/workernavigator/clearappbadge/index.md:54]:
```suggestion
```

File: files/en-us/web/api/workernavigator/setappbadge/index.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+{{Specifications}}
+
+## Browser compatibility
+
+{{Compat}}
+
+## See also
+
+- [Badging API](/en-US/docs/Web/API/Badging_API)
```
Comment (skyclouds2001) [inline files/en-us/web/api/workernavigator/setappbadge/index.md:57]:
```suggestion
```

**Review by teoli2003 — APPROVED** [review]
👍

**Pull request merged**